### PR TITLE
feat: support applying music mode constraints

### PR DESCRIPTION
**Pull request by brycetham**
This PR adds the necessary constraints and methods to support music mode on local audio tracks.

It adds the following APIs to LocalTrack:

- `applyConstraints`: useful for applying music mode constraints to the track, though [it doesn't work for audio constraints on Chrome](https://bugs.chromium.org/p/chromium/issues/detail?id=796964); still, I think it'll be useful for other browsers like Firefox, so that it doesn't need to make a new call to `getUserMedia`.
- `getConstraints`: not strictly necessary for music mode but useful for testing purposes.
- `getSettings`: as far as I know, this is the recommended way to get the device ID of the track, useful for calling `getUserMedia` again on Chrome.

### CODE REVIEW

**Comment by arun3528** [comment]
thanks for testing it out, The code base will change significantly with the beta version https://github.com/webex/webrtc-core/tree/beta

just FYI

**Comment by bbaldino** [comment]
lgtm.  @arun3528  should he do this against the beta branch instead?

**Comment by brycetham** [comment]
@bbaldino I think we're going to wait until the beta branch is merged in, and then merge this PR in after.

**Review by bbaldino — APPROVED** [review]

**Pull request merged**
